### PR TITLE
Removing most warnings due to units.

### PR DESCRIPTION
**Pull request by abotiamnot**
Fixes issue #472

### CODE REVIEW

**Comment by abotiamnot** [comment]
@dstansby Needs work! But this is the approach I'm thinking of. How does it look?

**Comment by abotiamnot** [comment]
This is quite confusing. Why aren't the units being returned?

**Comment by abotiamnot** [comment]
@dstansby I'm a bit confused how to fix that CircleCI error. Can you provide some help?

File: heliopy/data/custom/units_dict.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""
```
Comment (dstansby) [inline heliopy/data/custom/units_dict.py]: Could the code here go inside `heliopy/data/helpers` instead?

File: heliopy/data/custom/units_dict.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+-------------------
+Method to obtain the unit denoted by the strings
+inside the CDF files in the UNIT attribute.
+"""
+
+import astropy.units as u
+from collections import OrderedDict
+
+
+def get(unit_string):
```
Comment (dstansby) [inline heliopy/data/custom/units_dict.py]: 👍 I like this

File: heliopy/data/custom/units_dict.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    units = OrderedDict([('ratio', u.dimensionless_unscaled),
+                        ('#/cc', u.cm**-3),
+                        ('#/cm^3', u.cm**-3),
+                        ('cm^{-3}', u.cm**-3),
+                        ('ionic charge', u.electron),
```
Comment (dstansby) [inline heliopy/data/custom/units_dict.py]: Does `u.electron` give a negative number? If so this should be `u.electron`.

File: heliopy/data/custom/units_dict.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                        ('earth radii', u.earthRad),
+                        ('Re', u.earthRad),
+                        ('Degrees', u.deg),
+                        ('degrees', u.deg),
+                        ('#/{cc*(cm/s)^3}', (u.cm**3 * (u.cm / u.s)**3)**-1),
+                        ('Angle', u.deg),
```
Comment (dstansby) [inline heliopy/data/custom/units_dict.py]: Angles might not always be degrees (they could be radians), so best not to have this mapping in general and to do it on a case by case basis for each function instead.

File: heliopy/data/util.py [hunk 5]
```diff
@@ -232,9 +233,10 @@ def cdf_units(cdf_, keys=None, manual_units=None):
                 if key in manual_units:
                     continue
             unknown_unit = (cdf_[key].attrs['UNITS'])
-            message = "{} unit,{} column unknown".format(unknown_unit, key)
-            warnings.warn(message, Warning)
-            temp_unit = None
+            temp_unit = units_dict.get(unknown_unit)
+            if temp_unit is None:
+                message = "{} unit, {} key unknown".format(unknown_unit, key)
```
Comment (dstansby) [inline heliopy/data/util.py]: Could this say "The CDF provided units ({}) for key '{}' are unknown"

File: heliopy/data/custom/units_dict.py (hunk 4 above)
Comment (abotiamnot) [reply heliopy/data/custom/units_dict.py]: Got it!

File: heliopy/data/custom/units_dict.py (hunk 1 above)
Comment (abotiamnot) [reply heliopy/data/custom/units_dict.py]: Yep moved!

File: heliopy/data/custom/units_dict.py (hunk 3 above)
Comment (abotiamnot) [reply heliopy/data/custom/units_dict.py]: It doesn't give a negative number, I think we should define something else for this.

Comment (abotiamnot) [reply heliopy/data/custom/units_dict.py]: Is the ionic charge here referring to the charged state? Like O^- or H3O+?

**Comment by abotiamnot** [comment]
@dstansby I think 'Extracting CDF manually' warning shouldn't be removed. It is really important. For the rest, I think we need to do some bug fixes! Like, wind.swe_h1 has some useless columns which need to be removed.

**Pull request merged**